### DISCUSSION TITLE
KeyError: 'distribution' while running elastalert-create-index

### DISCUSSION THREAD

**Discussion by spasaintk** [jertel/elastalert2#487] [Q&A] [answered]
Hello,

I am entirely new to Elastalert2 so please excuse me if I'm missing something obvious. I've checked the documentation and discussions but I cannot find an answer to the issue I run in to.
I am attempting to run elastalert2 directly on the host. I am running a Debian 11 install with Python3.9 and I've cloned Elastalert2 from this git.  On running the 'elastalert-create-index' step from the manual and providing the required information (host, port, ssl, auth) I get presented with the following error:

elastalert-create-index
Enter Elasticsearch host: some.host
Enter Elasticsearch port: 9200
Use SSL? t/f: t
Verify TLS certificates? t/f: t
Enter optional basic-auth username (or leave blank): someuser
Enter optional basic-auth password (or leave blank): somepass
Enter optional Elasticsearch URL prefix (prepends a string to the URL of every request):
New index name? (Default elastalert_status)
Name of existing index to copy? (Default None)
Traceback (most recent call last):
  File "/usr/local/bin/elastalert-create-index", line 33, in <module>
    sys.exit(load_entry_point('elastalert2==2.2.1', 'console_scripts', 'elastalert-create-index')())
  File "/usr/local/lib/python3.9/dist-packages/elastalert2-2.2.1-py3.9.egg/elastalert/create_index.py", line 278, in main
    create_index_mappings(es_client=es, ea_index=index, recreate=args.recreate, old_ea_index=old_index)
  File "/usr/local/lib/python3.9/dist-packages/elastalert2-2.2.1-py3.9.egg/elastalert/create_index.py", line 24, in create_index_mappings
    if esinfo['distribution'] == "opensearch":
KeyError: 'distribution'

Does anyone know what is going wrong here?

Cheers,

**Comment by jertel** [comment, marked as answer]
@nbrownus didn't you make a change to handle this situation in your PR?

**Reply by jertel** [reply]
@spasaintk You must have pulled right after I merged PR #483 earlier today, which had a flaw that another reviewer pointed out but was never corrected. I was able to reproduce what you ran into. I've pushed the correction and so you should be able to pull latest and retry. Thanks for pointing out the issue.

**Reply by spasaintk** [reply]
Triple A, 5 star service! Thanks a lot, working like a charm on the latest pull.
